### PR TITLE
fix api tests and output

### PR DESCRIPTION
**Pull request by neolynx**
## Description of the Change

The ReposAPITestCopyPackage sporadically fails (i.e `\u003cEOL\u003e` (<EOL>) as filename).

This change fixes the implementation of apiReposCopyPackage, making sure the query parameter is read outside of `maybeRunTaskInBack`.

Also, check if there are packages to copy in ReposAPITestCopyPackage or fail with an error. 

This change also improves the error output in case a test fails, and enables s3 tests locally when running `make docker-system-tests`. Also, some lost tests were reenabled and others fixed.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -58,7 +58,10 @@ endif
 
 docker-test: install
 	go test -v -coverpkg="./..." -c -tags testruncli
-	PATH=$(BINPATH)/:$(PATH) APTLY_VERSION=$(VERSION) $(PYTHON) system/run.py --long $(TESTS) --coverage-dir $(COVERAGE_DIR) $(CAPTURE) $(TEST)
+	test -e aws.creds && . ./aws.creds; \
```
Comment (an0rak-dev) [inline Makefile:61]: Is this file automatically created during the `make docker-build-system-tests` (or any other dependant) target ?

File: system/t12_api/snapshots.py [hunk 2]
```diff
@@ -208,7 +213,7 @@ def check(self):
                  "Signing": DefaultSigningOptions,
             }
         )
-        self.check_equal(resp.json()['State'], TASK_SUCCEEDED)
+        self.check_task(task)
 
         self.check_equal(self.delete_task("/api/snapshots/" + snap2).json()['State'], TASK_FAILED)
         self.check_equal(self.delete_task("/api/snapshots/" + snap2, params={"force": "1"}).json()['State'], TASK_FAILED)
```
Comment (an0rak-dev) [inline system/t12_api/snapshots.py:219]: Why not replacing occurences of `check_equal(..., TASK_FAILED)` with `check_task_fail` in this file ?

**Review by an0rak-dev — CHANGES_REQUESTED** [review]

File: Makefile (hunk 1 above)
Comment (neolynx) [reply Makefile:61]: no, the file has to be place there manually, if it is there s3 tests will be enabled..

File: system/t12_api/snapshots.py (hunk 2 above)
Comment (neolynx) [reply system/t12_api/snapshots.py]: fixed, thanks !  :)

File: Makefile (hunk 1 above)
Comment (an0rak-dev) [reply Makefile]: Should we not indicate it in the ReadMe or in the project's wiki ?

Comment (neolynx) [reply Makefile]: tthis is true, alongside with openstack swift and azure testing...

I will move it to a next PR, has nothing to do in here.

Comment (an0rak-dev) [reply Makefile]: Sounds great.

Let me know when you have moved the aws.creds part, and this PR will be Ok for me 👌🏻

Comment (neolynx) [reply Makefile]: I moved it our of this PR

**Review by an0rak-dev — APPROVED** [review]

**Pull request merged**